libandroid-navigation-ui/src/main/java/org/maplibre/navigation/android/navigation/ui/v5/NavigationViewRouter.java

```diff
@@ -118,7 +120,7 @@ private void cacheRouteOptions(RouteOptions routeOptions) {
   private void cacheRouteDestination() {
     boolean hasValidCoordinates = routeOptions != null && !routeOptions.getCoordinates().isEmpty();
     if (hasValidCoordinates) {
-      List<Point> coordinates = routeOptions.getCoordinates();
+      List<Point> coordinates = toMapLibrePoints(routeOptions.getCoordinates());
```

Comment:
This converting is very annoying. We now having three `Point` types

- MapLibre (Android)
- MapLibre Navigation (generic KMP)
- Mapbox

But as mentioned, we need to decide how and where we want to have the generic GeoJson stuff. Changing this before this, make multiple effort and work.